lib/vector-core/buffers/src/lib.rs

```diff
@@ -205,3 +223,42 @@ impl<T, S: Sink<T> + Unpin> Sink<T> for DropWhenFull<S> {
         self.project().inner.poll_close(cx)
     }
 }
+
+// The BlockWhenFull wrapper is used for instrumentation purposes
```

Comment:
Good comment hygiene nit: you might consider saying _why_ `BlockWhenFull` exists rather than what it is. Why gives context around the struct, helps future readers understand if it can be changed and under what circumstances. 

I think this could also be a rustdoc proper.